R/mod_hierarchical_count_table.R

```diff
@@ -23,9 +47,159 @@ EC <- poc( # nolint
       NO_POP_ROWS = "Population dataset has 0 rows",
       GRP_CLASH = "Group selection cannot be used in hierarchy"
     )
+  ),
+  VAL = poc(
+    SPECIAL_CHAR = "\u001D" # For naming and processing hierarchy levels
   )
 )
 
+#' Create analysis data in the form of an ADaM ADTTE dataset
+#'
+#' @param event_df `data.frame`
+#' A data frame containing the event data. It should have columns corresponding to subjects,
+#' hierarchy levels, and group variables.
+#'
+#' @param pop_df `data.frame`
+#' A data frame containing the population data. It must have columns corresponding to subjects and group variables.
+#'
+#' @param hierarchy `character(1|2)`
+#' A character vector of column names from `event_df` to use as the hierarchy. Can be one or two levels.
+#'
+#' @param group_var `character(1)`
+#' A string representing the column name in `pop_df` used for grouping the population data.
+#'
+#' @param subjid_var `character(1)`
+#' A string representing the subject identifier column name. This column must be present in both `event_df` and
+#' `pop_df`.
+#'
+#' @param origin_date_var `character(1)`
+#' A string representing the column name in `pop_df` holding the origin date.
+#'
+#' @param censor_date_var `character(1)`
+#' A string representing the column name in `pop_df` holding the censor date.
+#'
+#' @param event_date_var `character(1)`
+#' A string representing the column name in `event_df` holding the event date.
+#'
+#' @param param_val `character(1)`
+#' A string representing the value to assign to `PARAM` column. When `recurse` is `TRUE` hierarchy level is prepended
+#' with `L# ` where `#` is `0` for the top level, `1` for the next level, and so on.
+#'
+#' @param paramcd_val `character(1)`
+#' A string representing the value to assign to `PARAMCD` column. When `recurse` is `TRUE` hierarchy level is prepended
+#' with `L#_` where `#` is `0` for the top level, `1` for the next level, and so on.
+#'
+#' @param recurse `logical(1)`
+#' A logical indicating whether to prepend data from calling the function again for higher hierarchy levels.
+#'
+#' @return A data frame in the form of an ADaM ADTTE, with the addition of hierarchy columns and hierarchy level.
+#'
+#' @keywords internal
+create_adtte <- function(event_df,
+                         pop_df,
+                         hierarchy = character(),
+                         group_var,
+                         subjid_var,
+                         origin_date_var,
+                         censor_date_var,
+                         event_date_var,
+                         param_val = "Time to Event",
+                         paramcd_val = "EVENT",
+                         recurse = TRUE) {
```

Comment:
When is it possible that recurse is FALSE? As far as I can see all calls have recurse = TRUE. Recursiveness ends when there are no more hierarchy levels. Unsure why this is required.